HPXMLtoOpenStudio/resources/hpxml_defaults.rb

```diff
@@ -444,6 +444,78 @@ def self.apply_building(hpxml_bldg, epw_file)
+        hpxml_bldg.site.ground_diffusivity_isdefaulted = true
+      elsif hpxml_bldg.site.soil_type == HPXML::SiteSoilTypeSilt || hpxml_bldg.site.soil_type == HPXML::SiteSoilTypeClay
+        if hpxml_bldg.site.moisture_type == HPXML::SiteSoilMoistureTypeDry
+          hpxml_bldg.site.ground_conductivity = 0.2889 # Btu/hr-ft-F
+          hpxml_bldg.site.ground_diffusivity = 0.0120 # ft^2/hr
+        elsif hpxml_bldg.site.moisture_type == HPXML::SiteSoilMoistureTypeWet
+          hpxml_bldg.site.ground_conductivity = 0.9821 # Btu/hr-ft-F
+          hpxml_bldg.site.ground_diffusivity = 0.0194 # ft^2/hr
+        elsif hpxml_bldg.site.moisture_type == HPXML::SiteSoilMoistureTypeMixed
+          hpxml_bldg.site.ground_conductivity = ((0.2889 + 0.9821) / 2.0).round(4) # Btu/hr-ft-F
+          hpxml_bldg.site.ground_diffusivity = ((0.0120 + 0.0194) / 2.0).round(4) # ft^2/hr
+        end
+        hpxml_bldg.site.ground_conductivity_isdefaulted = true
+        hpxml_bldg.site.ground_diffusivity_isdefaulted = true
+      elsif hpxml_bldg.site.soil_type == HPXML::SiteSoilTypeLoam
+        hpxml_bldg.site.ground_conductivity = 1.2132 # Btu/hr-ft-F
+        hpxml_bldg.site.ground_diffusivity = 0.0353 # ft^2/hr
+
+        hpxml_bldg.site.ground_conductivity_isdefaulted = true
+        hpxml_bldg.site.ground_diffusivity_isdefaulted = true
+      elsif hpxml_bldg.site.soil_type == HPXML::SiteSoilTypeGravel
+        if hpxml_bldg.site.moisture_type == HPXML::SiteSoilMoistureTypeDry
+          hpxml_bldg.site.ground_conductivity = 0.2311 # Btu/hr-ft-F
+          hpxml_bldg.site.ground_diffusivity = 0.0097 # ft^2/hr
+        elsif hpxml_bldg.site.moisture_type == HPXML::SiteSoilMoistureTypeWet
+          hpxml_bldg.site.ground_conductivity = 1.0399 # Btu/hr-ft-F
+          hpxml_bldg.site.ground_diffusivity = 0.0291 # ft^2/hr
+        elsif hpxml_bldg.site.moisture_type == HPXML::SiteSoilMoistureTypeMixed
+          hpxml_bldg.site.ground_conductivity = ((0.2311 + 1.0399) / 2.0).round(4) # Btu/hr-ft-F
+          hpxml_bldg.site.ground_diffusivity = ((0.0097 + 0.0291) / 2.0).round(4) # ft^2/hr
+        end
+        hpxml_bldg.site.ground_conductivity_isdefaulted = true
+        hpxml_bldg.site.ground_diffusivity_isdefaulted = true
+      elsif hpxml_bldg.site.soil_type == HPXML::SiteSoilTypeUnknown
+        hpxml_bldg.site.ground_conductivity = 1.0 # Btu/hr-ft-F
+        hpxml_bldg.site.ground_diffusivity = 0.0208 # ft^2/hr
+
+        hpxml_bldg.site.ground_conductivity_isdefaulted = true
+        hpxml_bldg.site.ground_diffusivity_isdefaulted = true
+      end
+    elsif hpxml_bldg.site.ground_conductivity.nil?
+      hpxml_bldg.site.ground_conductivity = 1.0 # Btu/hr-ft-F
+      hpxml_bldg.site.ground_conductivity_isdefaulted = true
+    elsif hpxml_bldg.site.ground_diffusivity.nil?
+      hpxml_bldg.site.ground_diffusivity = 0.0208 # ft^2/hr
+      hpxml_bldg.site.ground_diffusivity_isdefaulted = true
+    end
```

Comment:
Seems like this can be simplified to:

```suggestion
      end
    end
    if hpxml_bldg.site.ground_conductivity.nil?
      hpxml_bldg.site.ground_conductivity = 1.0 # Btu/hr-ft-F
      hpxml_bldg.site.ground_conductivity_isdefaulted = true
    elsif hpxml_bldg.site.ground_diffusivity.nil?
      hpxml_bldg.site.ground_diffusivity = 0.0208 # ft^2/hr
      hpxml_bldg.site.ground_diffusivity_isdefaulted = true
    end
```

Reply:
Also, does it really make sense for diffusivity and conductivity to be defaulted independently? I.e., if the user enters two different conductivities, they will have the same default diffusivity -- but the two properties are related.

Reply:
Yes, they're related. So in that example where there's two different conductivities and we default the diffusivity, we're basically assuming a change in either density or specific heat that perfectly offsets the difference.

I'd vote for either defaulting both or neither as Scott suggested, rather than allowing both independently.

Reply:
I didn't mean to suggest we should restrict the inputs. I think it'd be a poor approach to force a user to have to enter a ground diffusivity when all they want to enter is a different ground conductivity.

Rather, maybe we should just preserve the relationship between the two. E.g., conductivity/diffusivity = 1.0/0.0208. If the user enters a conductivity of 2, they get a diffusivity of 0.0208*2.

Reply:
Preserving the 1.0/0.0208 relationship if either conductivity or diffusivity not provided here: https://github.com/NREL/OpenStudio-HPXML/pull/1391/commits/54e1dd0436c6d1fac441238535135625f2e64f6c